src/wallet/bdb.h

```diff
@@ -130,9 +125,13 @@ class BerkeleyDatabase
      */
     bool Backup(const std::string& strDest) const;
 
-    /** Make sure all changes are flushed to disk.
+    /** Make sure all changes are flushed to database file.
+     */
+    void Flush();
+    /** Flush to the database file and close the database.
+     *  Also close the environment if no other databases are open in it.
      */
-    void Flush(bool shutdown);
+    void Close();
```

Comment:
I would prefer to keep the naming and move to make the functions actually do as the names suggest. Since the goal is to have a standard interface, changing the names to match the craziness we have for BDB seems counterproductive. For example, in #19077, Flush and Close do work as you would expect.

I think many of these issues stem from the fact that we allow multiple BerkeleyDatabases for one BerkeleyEnvironment. I think if we stopped doing that, we would be able to make Flush and Close behave in the ways that we expect.